src/main/java/seedu/address/logic/parser/InternshipEditCommandParser.java

```diff
@@ -0,0 +1,110 @@
+                PREFIX_CONTACT_NUMBER, PREFIX_LOCATION, PREFIX_STATUS, PREFIX_DESCRIPTION, PREFIX_ROLE);
+
+        EditInternshipDescriptor editPersonDescriptor = new EditInternshipDescriptor();
+
+        if (argMultimap.getValue(PREFIX_COMPANY).isPresent()) {
+            editPersonDescriptor.setCompanyName(InternshipParserUtil.parseCompanyName(argMultimap
+                    .getValue(PREFIX_COMPANY).get()));
+        }
+        if (argMultimap.getValue(PREFIX_CONTACT_NAME).isPresent()) {
+            editPersonDescriptor.setContactName(InternshipParserUtil.parseContactName(argMultimap
+                    .getValue(PREFIX_CONTACT_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_CONTACT_NUMBER).isPresent()) {
+            editPersonDescriptor.setContactNumber(InternshipParserUtil.parseContactNumber(argMultimap
+                    .getValue(PREFIX_CONTACT_NUMBER).get()));
+        }
+        if (argMultimap.getValue(PREFIX_CONTACT_EMAIL).isPresent()) {
+            editPersonDescriptor.setContactEmail(InternshipParserUtil.parseContactEmail(argMultimap
+                    .getValue(PREFIX_CONTACT_EMAIL).get()));
+        }
+        if (argMultimap.getValue(PREFIX_LOCATION).isPresent()) {
+            editPersonDescriptor.setLocation(InternshipParserUtil.parseLocation(argMultimap
+                    .getValue(PREFIX_LOCATION).get()));
+        }
+        if (argMultimap.getValue(PREFIX_STATUS).isPresent()) {
+            editPersonDescriptor.setApplicationStatus(InternshipParserUtil.parseStatus(argMultimap
+                    .getValue(PREFIX_STATUS).get()));
+        }
+        if (argMultimap.getValue(PREFIX_DESCRIPTION).isPresent()) {
+            editPersonDescriptor.setDescription(InternshipParserUtil.parseDescription(argMultimap
+                    .getValue(PREFIX_DESCRIPTION).get()));
+        }
+        if (argMultimap.getValue(PREFIX_ROLE).isPresent()) {
+            editPersonDescriptor.setRole(InternshipParserUtil.parseRole(argMultimap
+                    .getValue(PREFIX_ROLE).get()));
+        }
+
+        if (!editPersonDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(InternshipEditCommand.MESSAGE_NOT_EDITED);
+        }
+
+        return new InternshipEditCommand(index, editPersonDescriptor);
+    }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Tag>} containing zero tags.
+     */
+    private Optional<Set<Tag>> parseTagsForEdit(Collection<String> tags) throws ParseException {
+        assert tags != null;
+
+        if (tags.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> tagSet = tags.size() == 1 && tags.contains("") ? Collections.emptySet() : tags;
+        return Optional.of(ParserUtil.parseTags(tagSet));
```

Comment:
Should these functions still be here for Internships because we don't have tags anymore